src/general/g.citation/g.citation.py

```diff
@@ -188,6 +188,35 @@ def clean_line_item(text):
     return text
 
 
+def get_datetime_from_documentation(text):
+    """Extract year from text containing SVN date entry
+    >>> text = "  Latest change: Monday Jun 28 11:54:09 2021 in commit: 1cfc0af029a35a5d6c7dae5ca7204d0eb85dbc55"
+    >>> get_datetime_from_documentation(text)
+    datetime.datetime(2022, 9, 18, 23, 55, 9)
+    """
+    datetime_capture = (
+        r"^  (Latest change: )(.*)( in commit: ).*"
+        )
+    match = re.search(datetime_capture, text, re.MULTILINE | re.DOTALL | re.IGNORECASE)
+    if match:
+        date_format = "%A %b %d %H:%M:%S %Y"
+    else:
+        datetime_capture = (
```

Comment:
1. I'm tested this PR with GRASS GIS 8.3.dev version compiled from Git source (last change in module manual is retrieved from commit), **all tests pass** with python3.{8..11}

2. With GRASS GIS 8.2.0 version compiled from tarball without applied this patch https://github.com/OSGeo/grass/releases/download/8.2.0/core_modules_with_last_commit.patch  (last change in module manual is retrieved from the file) **all tests fail** with python3.{8..11}. Same result as @neteler.

Due this line code https://github.com/OSGeo/grass/blob/f9a30a386c46ea1f281a14ad5b9570aa84a078a7/utils/mkhtml.py#L166 which return date time with following format `Wed Oct  5 05:23:22 2022` with 2 white spaces between `Oct\s\s5` (`\s` is white space), regular expression doesn't catch this date time format.

Correct regular expression should be:

`r"^  (Accessed: ).*([A-Z][a-z]{2} [A-Z][a-z]{2}\s\s[0-9]{1,2} [0-9]{2}:[0-9]{2}:[0-9]{2} [0-9]{4}).*"` with two white spaces `\s` between month and day.

Anyway, I suggest that we use the same date time format (`"%A %b %d %H:%M:%S %Y"`) here https://github.com/OSGeo/grass/blob/f9a30a386c46ea1f281a14ad5b9570aa84a078a7/utils/mkhtml.py#L166 as in other cases.

Example:

```python
In [58]: import os

In [59]: from datetime import datetime

In [60]: datetime_format = "%A %b %d %H:%M:%S %Y"

In [61]: datetime.fromtimestamp(os.path.getmtime("/tmp")).strftime(datetime_format)
Out[61]: 'Wednesday Oct 05 06:42:36 2022'
```

Reply:
Thanks, @tmszi  for catching the issue! I tried to address it on the addon side for now, by filtering one or more spaces between month and day ([75f8e72](https://github.com/OSGeo/grass-addons/pull/677/commits/75f8e7253194fd201df30dedcf9f4f3c85fba1ec)). I agree that a consistent format in the manuals would be preferable.
And thanks, @wenzeslaus. It seems I messed up the PR again when I tried to include #806 to get the CI working for this PR... Will try your suggestion to cleanup the mess later today!